server/src/domain/media/media.service.ts

```diff
@@ -186,12 +186,18 @@ export class MediaService {
     }
 
     const isTargetVideoCodec = videoStream.codecName === ffmpegConfig.targetVideoCodec;
-    const isTargetAudioCodec = audioStream.codecName === ffmpegConfig.targetAudioCodec;
     const isTargetContainer = ['mov,mp4,m4a,3gp,3g2,mj2', 'mp4', 'mov'].includes(containerExtension);
+    const isTargetAudioCodec = audioStream != null && audioStream.codecName === ffmpegConfig.targetAudioCodec;
```

Comment:
Does this always transcode if there's no audio stream? I think it makes sense to not transcode if the video streams match since there's no format change on transcoding. I also don't know what FFmpeg does if you specify an output audio codec when there's no input audio.

Reply:
It seems like this should be "is target codec = no audio stream or audio stream matches target"

Reply:
> It seems like this should be "is target codec = no audio stream or audio stream matches target"

That makes sense! Made the change accordingly